### ISSUE TITLE
Kill assists

### ISSUE THREAD

**Issue by ClobberXD** [MT-CTF/capturetheflag#184]
### Proposal
There are endless situations where one player sacrifices their life to deal severe damage to an enemy, and another player finishes them off. In cases like these, all players who deal more than `x` hp of damage should be awarded `y%` of the kill reward the killer gets. Kill assists will have its own column in the `/rankings` table and match summaries, like bounty kills.

(Values of `x` and `y` to be decided later)

### Implementation
- Add new field kill assists to `ctf_stats`.
- Track PvP damage, if player deals out more than `x` hp to an enemy, add their name to a list which keeps track of kill assists. Each player will their own list to store names of their kill assist-ers.
- Once player has been killed, award `y%` of kill reward to all players in the list.
- Increment those players' `kill assists` field in `ctf_stats`.

**Comment by ClobberXD** [comment]
If it's a good idea, I'll start working on it. Where do the kill assist lists of the players go? I'm thinking `ctf_stats.kill_assists[name]`, where `name` is the victim's name

**Comment by Coder12a** [comment]
How about give the player who hit the enemy player first all the points.

**Comment by AKryukov92** [comment]
> How about give the player who hit the enemy player first all the points.

We want to encourage kills, not ninja first-hitters. Getting kill much more important than striking first. Thats why we give score to killers.

**Comment by afshaan4** [comment]
wont work coz then you'd have situations where a player that barely helped in the kill gets all the points

**Comment by ClobberXD** [comment]
I closed the PR a few months ago because while the current concept is okay, I wanted (and still want) a proper API for registering stats, before kill assists is added. I forgot all about this because this was closed (for whatever reason) in spite of the PR remaining unmerged. I'm adding this to the milestone.

Also, this depends on a non-existent PR to implement the API, which in turn depends on #518 being merged (otherwise Imma anyway have to redo the whole PR once that PR gets merged, because merge conflicts).
